src/GNDStk/child_t/test/operator.test.cpp

```diff
@@ -0,0 +1,227 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+using namespace njoy::GNDStk;
+
+
+// converterA
+struct converterA {
+   template<class FROM, class TO>
+   void operator()(const FROM &from, TO &to) const { convert(from,to); }
+};
+
+// converterB
+struct converterB {
+   template<class FROM, class TO>
+   void operator()(const FROM &from, TO &to) const { convert(from,to); }
+};
+
+
+// is_void(child_t)
+template<find FIND>
+bool is_void(const child_t<void,FIND>)
+{ return true; }
+
+template<class TYPE, find FIND, class CONVERTER>
+bool is_void(const child_t<TYPE,FIND,CONVERTER>)
+{ return false; }
+
+
+// is_int(child_t)
+template<find FIND, class CONVERTER>
+bool is_int(const child_t<int,FIND,CONVERTER>)
+{ return true; }
+
+template<class TYPE, find FIND, class CONVERTER>
+bool is_int(const child_t<TYPE,FIND,CONVERTER>)
+{ return false; }
+
+
+// is_float(child_t)
+template<find FIND, class CONVERTER>
+bool is_float(const child_t<float,FIND,CONVERTER>)
+{ return true; }
+
+template<class TYPE, find FIND, class CONVERTER>
+bool is_float(const child_t<TYPE,FIND,CONVERTER>)
+{ return false; }
+
+
+// is_double(child_t)
+template<find FIND, class CONVERTER>
+bool is_double(const child_t<double,FIND,CONVERTER>)
+{ return true; }
+
+template<class TYPE, find FIND, class CONVERTER>
+bool is_double(const child_t<TYPE,FIND,CONVERTER>)
+{ return false; }
+
+
+// is_converterA(child_t)
+template<class TYPE, find FIND>
+bool is_converterA(const child_t<TYPE,FIND,converterA>)
+{ return true; }
+
+template<class TYPE, find FIND>
+bool is_converterA(const child_t<TYPE,FIND,converterB>)
+{ return false; }
+
+
+// is_converterB(child_t)
+template<class TYPE, find FIND>
+bool is_converterB(const child_t<TYPE,FIND,converterB>)
+{ return true; }
+
+template<class TYPE, find FIND>
+bool is_converterB(const child_t<TYPE,FIND,converterA>)
+{ return false; }
+
+
+// is_one(child_t)
+template<class TYPE, class CONVERTER>
+bool is_one(const child_t<TYPE,find::one,CONVERTER>)
+{ return true; }
+
+template<class TYPE, class CONVERTER>
+bool is_one(const child_t<TYPE,find::all,CONVERTER>)
+{ return false; }
+
+
+// is_all(child_t)
+template<class TYPE, class CONVERTER>
+bool is_all(const child_t<TYPE,find::all,CONVERTER>)
+{ return true; }
+
+template<class TYPE, class CONVERTER>
+bool is_all(const child_t<TYPE,find::one,CONVERTER>)
+{ return false; }
+
+
+
+// ------------------------
+// SCENARIO
+// ------------------------
+
+SCENARIO("Testing GNDStk child_t operators") {
+
+   // ------------------------
+   // -child_t
+   // ------------------------
+
+   GIVEN("A child_t<type,one|all,converter>") {
```

Comment:
So these test the functionality just fine, but they don't help me to understand how to use the library. I don't want you to change this, but in the future, please construct examples at least something closer to what we will actually do with GNDS

Reply:
The referenced material is from a brand-new test code, that tests some obscure features that I put in mainly because I anticipate possible interest from future developers, perhaps more so than from end users. This test *is* basically made to just mechanically test the features. In a real code, I'd expect somebody to use a mixed bag of features from here and there - all over the code base. Really, I think the bigger discussion here would be about a suite of tests, versus a series of examples (say, in a tutorial) geared towards users. Tests and examples are typically handled differently, even if perhaps in some ideal world they could be one and the same.

Reply:
I agree. Having a series of examples is a very good idea. Something that we need for people to start using this library. It would be easy to develop the series of examples; anytime someone asks how to do something, we add another example. Can be developed over time.